src/Datagrid/SimplePager.php

```diff
@@ -66,22 +66,22 @@ public function getNbResults()
         return $n;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function getResults($hydrationMode = null)
     {
         if ($this->results) {
             return $this->results;
         }
 
-        $this->results = $this->getQuery()->execute([], $hydrationMode);
-        $this->thresholdCount = count($this->results);
-        if (count($this->results) > $this->getMaxPerPage()) {
-            $this->haveToPaginate = true;
+        $this->results        = $this->getQuery()->execute([], $hydrationMode);
+        $resultCount          = \count($this->results);
+        $this->thresholdCount = $resultCount;
 
-            if ($this->results instanceof ArrayCollection) {
-                $this->results = new ArrayCollection($this->results->slice(0, $this->getMaxPerPage()));
-            } else {
-                $this->results = new ArrayCollection(array_slice($this->results, 0, $this->getMaxPerPage()));
-            }
```

Comment:
I see this as a BC break, maybe I am wrong but I would rather change the Interface comment about return type than this.

Reply:
I think it's better to stick to `array` because `DatagridInterface::getResults` is also promising an `array`. I've checked implementations of `Pager` in ORM/PHPCR/Mongo/Search and they will return `array`

By changing return-type in `PagerInterface`, we'll be forced to change type in `DatagridInterface`, because it is direct consumer of `Pager` and it should respect that `SimplePager` will return `Collection` or as another solution without changing any annotation is to handle this `Collection` in `Datagrid` implementation, which still won't solve issue with inconsistency of return-types

Reply:
I get it but we can’t just change the return value, what if someone is now using the ArrayCollection? I think we can just add deprecation for now and fix it on next major. cc @sonata-project/contributors

Reply:
@kunicmarko20 I'm fine with that, if there are no rejections